godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -348,139 +363,300 @@ macro_rules! impl_float_vector_component_fns {
         ($($comp:ident),*)
     ) => {
         impl $Vector {
-            pub fn lerp(self, other: Self, weight: $Scalar) -> Self {
-                Self::new(
+            /// Returns a vector with the given components.
+            pub const fn new($($comp: $Scalar),*) -> Self {
+                Self {
                     $(
-                        self.$comp.lerp(other.$comp, weight)
+                        $comp
                     ),*
-                )
+                }
             }
 
-            pub fn bezier_derivative(self, control_1: Self, control_2: Self, end: Self, t: $Scalar) -> Self {
-                $(
-                    let $comp = self.$comp.bezier_derivative(control_1.$comp, control_2.$comp, end.$comp, t);
-                )*
+            /// Returns a new vector with all components set to `v`.
+            pub const fn splat(v: $Scalar) -> Self {
+                Self {
+                    $(
+                        $comp: v
+                    ),*
+                }
+            }
 
-                Self::new($($comp),*)
+            /// Returns a new vector with all components in absolute values (i.e. positive or
+            /// zero).
+            #[inline]
+            pub fn abs(self) -> Self {
+                Self::from_glam(self.to_glam().abs())
             }
 
-            pub fn bezier_interpolate(self, control_1: Self, control_2: Self, end: Self, t: $Scalar) -> Self {
-                $(
-                    let $comp = self.$comp.bezier_interpolate(control_1.$comp, control_2.$comp, end.$comp, t);
-                )*
+            /// Returns a new vector with all components clamped between the components of `min` and `max`.
+            ///
+            /// Panics
+            /// Panics if `min` > `max`, `min` is NaN, or `max` is NaN.
+            #[inline]
+            pub fn clamp(self, min: Self, max: Self) -> Self {
+                Self::from_glam(self.to_glam().clamp(min.to_glam(), max.to_glam()))
+            }
 
-                Self::new($($comp),*)
+            /// Returns the length (magnitude) of this vector.
+            #[inline]
+            pub fn length(self) -> real {
+                // does the same as glam's length() but also works for integer vectors
+                (self.length_squared() as real).sqrt()
```

Comment:
Instead of `as real`, please use the `RealConv` functions.

Reply:
`length_squared()` returns `i32` on integer vectors so I'm not sure which `RealConv ` function to use

Reply:
I see, thanks! :+1: